### PR TITLE
Fixed dark mode bugs in the availability section

### LINKED ISSUE #8628: [CAL-1612] Dark mode bugs availability
Found a bug? Please fill out the sections below. 👍

### Issue Summary

1) set as default on hover 

Screenshot 2023-05-03 at 1 12 28 AM

2) delete icon on hover

Screenshot 2023-05-03 at 1 12 52 AM

should be like this
Screenshot 2023-05-03 at 1 18 13 AM

3) Delete icon on hover should be red

Screenshot 2023-05-03 at 1 13 43 AM

### Steps to Reproduce

1) go to http://localhost:3000/availability
2) Open any availability item

<sub>[CAL-1612](https://linear.app/calcom/issue/CAL-1612/dark-mode-bugs-availability)</sub>

### PR DESCRIPTION
**Pull request by MaheshB77**
## What does this PR do?
- Fixed dark mode bugs in the availability section

Fixes #8628 

1. "Set as default" on hover was getting a white background, **FIXED**
chrome-capture-2023-4-3

2. Delete icon on hover was not showing the proper tooltip and background color, **FIXED**
issue-2

3. Delete icon on hover should be red, **FIXED**
issue-3

**Environment**: Production

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?
- Can be tested by visiting **/availability** section

### CODE REVIEW

File: apps/web/pages/availability/[schedule].tsx [hunk 1]
```diff
@@ -188,8 +188,8 @@ export default function Availability() {
       }
       CTA={
         <div className="flex items-center justify-end">
-          <div className="hidden items-center rounded-md px-2 sm:flex sm:hover:bg-gray-100">
-            <Skeleton as={Label} htmlFor="hiddenSwitch" className="mt-2 cursor-pointer self-center pr-2 ">
```
Comment (Udit-takkar) [inline apps/web/pages/availability/[schedule].tsx:192]: don't remove cursor-pointer

File: apps/web/pages/availability/[schedule].tsx [hunk 2]
```diff
@@ -212,7 +212,7 @@ export default function Availability() {
                 aria-label={t("delete")}
                 className="hidden sm:inline"
                 disabled={schedule?.isLastSchedule}
-                tooltip={t("requires_at_least_one_schedule")}
+                tooltip={t("delete")}
```
Comment (Udit-takkar) [inline apps/web/pages/availability/[schedule].tsx:215]: don't change this tooltip

Comment (MaheshB77) [reply apps/web/pages/availability/[schedule].tsx:215]: > don't change this tooltip

Sure @Udit-takkar will update the PR shortly

Comment (MaheshB77) [reply apps/web/pages/availability/[schedule].tsx]: @Udit-takkar Updated the PR accordingly, please check

**Review by Udit-takkar — APPROVED** [review]

**Review by keithwillcode — DISMISSED** [review]
Looks like the message to delete a 2nd working hours is saying "You are required to have at least one schedule."

**Comment by MaheshB77** [comment]
> Looks like the message to delete a 2nd working hours is saying "You are required to have at least one schedule."

Yes @keithwillcode, I did remove that message in my 1st commit but reverted back as per @Udit-takkar 's suggestion to keep it as it is

**Comment by Udit-takkar** [comment]
@MaheshB77 actually we have to show this tooltip to tell the user why delete icon is disabled when you try to delete one and only availability.  just add a condition if schedule?.isLastSchedule then show requires_at_least_one_schedule else delete tooltip message

**Comment by MaheshB77** [comment]
> 

That makes sense @Udit-takkar , I will update the PR accordingly, Thanks 🥂.

**Comment by MaheshB77** [comment]
> @MaheshB77 actually we have to show this tooltip to tell the user why delete icon is disabled when you try to delete one and only availability. just add a condition if schedule?.isLastSchedule then show requires_at_least_one_schedule else delete tooltip message

@Udit-takkar @keithwillcode Can you guys check now, I have updated the tooltip based on **schedule?.isLastSchedule** flag

**Comment by Udit-takkar** [comment]
Just need to confirm one thing from @sean-brydon 

1)
Screenshot 2023-05-04 at 1 11 34 AM

2) This is current ui when you hover over disabled delete button but i think 1 is better. was this intentional ?

Screenshot 2023-05-04 at 1 14 44 AM

**Comment by Udit-takkar** [comment]
I have pushed this now

Screenshot 2023-05-04 at 1 43 54 AM

**Review by sean-brydon — APPROVED** [review]
Code LGTM

**Comment by zeeshanbhati** [comment]
@MaheshB77 , Just a suggestion that don't you think the hover effect on the set as default button should be bg-subtle , I checked the toggle button in event-type/[id] it has a hover effect on bg-subtle and cursor pointer should stay there , that toggle should work on the title click as well

SS From ```event-type/[id]```
image

**Comment by sean-brydon** [comment]
> @MaheshB77, @Udit-takkar Just a suggestion that don't you think the hover effect on the set as default button should be bg-subtle, I checked the toggle button in event-type/[id] it has a hover effect of bg-subtle and cursor pointer should stay there, that toggle should look like that it works on the title click as well
> 
> SS From `event-type/[id]` image

Ill fix this in a follow up PR thank you :)

**Comment by zeeshanbhati** [comment]
> > @MaheshB77, @Udit-takkar Just a suggestion that don't you think the hover effect on the set as default button should be bg-subtle, I checked the toggle button in event-type/[id] it has a hover effect of bg-subtle and cursor pointer should stay there, that toggle should look like that it works on the title click as well
> > SS From `event-type/[id]` image
> 
> Ill fix this in a follow up PR thank you :)

@sean-brydon #8680 already does that, you can cherry-pick that commit into this PR :)

**Comment by sean-brydon** [comment]
> > > @MaheshB77, @Udit-takkar Just a suggestion that don't you think the hover effect on the set as default button should be bg-subtle, I checked the toggle button in event-type/[id] it has a hover effect of bg-subtle and cursor pointer should stay there, that toggle should look like that it works on the title click as well
> > > SS From `event-type/[id]` image
> > 
> > 
> > Ill fix this in a follow up PR thank you :)
> 
> @sean-brydon #8680 already does that, you can cherry-pick that commit into this PR :)

Thank you! Me and @Jaibles are reworking the switch component so will likely keep this in a separate PR. Thanks again!

**Comment by Udit-takkar** [comment]
@MaheshB77 can you fix merge conflicts ?

**Comment by MaheshB77** [comment]
> @MaheshB77 can you fix merge conflicts ?

@Udit-takkar Done, can you please merge this PR now 🙂

File: packages/ui/components/button/Button.tsx [hunk 3]
```diff
@@ -115,19 +115,14 @@ const buttonClasses = cva(
           "hover:bg-subtle focus-visible:bg-subtle focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset focus-visible:ring-empthasis",
       }),
       // Destructive variants
-      {
```
[the remark this answers is not in the document]
Comment (Udit-takkar) [reply packages/ui/components/button/Button.tsx:118]: @PeerRich 

https://github.com/calcom/cal.com/pull/8630#issuecomment-1533633864

**Pull request merged**